code-examples/observer-demo/src/EconomistObserverDemo.java

```diff
@@ -0,0 +1,107 @@
+import java.sql.Timestamp;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+
+public class EconomistObserverDemo {
+    public static void main(String[] args) {
+        Economist economist = new Economist();
+        Subscriber onlineSubscriber01 = new OnlineSubscription(economist);
+        Subscriber mobileSubscriber01 = new MobileSubscription(economist);
+        Subscriber physicalSubscriber01 = new PhysicalSubscription(economist);
+        Subscriber onlineSubscriber02 = new OnlineSubscription(economist);
+        Subscriber mobileSubscriber02 = new MobileSubscription(economist);
+        Subscriber physicalSubscriber02 = new PhysicalSubscription(economist);
+        Subscriber onlineSubscriber03 = new OnlineSubscription(economist);
+        Subscriber mobileSubscriber03 = new MobileSubscription(economist);
+        Subscriber physicalSubscriber03 = new PhysicalSubscription(economist);
+        Subscriber onlineSubscriber04 = new OnlineSubscription(economist);
+        Subscriber mobileSubscriber04 = new MobileSubscription(economist);
+        Subscriber physicalSubscriber04 = new PhysicalSubscription(economist);
+        Subscriber onlineSubscriber05 = new OnlineSubscription(economist);
+        Subscriber mobileSubscriber05 = new MobileSubscription(economist);
+        Subscriber physicalSubscriber05 = new PhysicalSubscription(economist);
+        for (int i = 0; i < 15; i++) {
+            economist.onEvent(new Event());
+            try {
+                int waitTime = 3000;
+                Thread.sleep(waitTime);
+            } catch (InterruptedException e) {
+                System.out.println(e);
+            }
+        }
+    }
+}
+
+class Event {
+    private static final SimpleDateFormat sdf = new SimpleDateFormat("yyyy.MM.dd.HH.mm.ss");
+
+    Event() {
+        Date date = new Date();
+        System.out.println("\nA new set of reviews for the subscribers of magazines at: " + new Timestamp(date.getTime()));
+    }
+}
+
+class Economist {
+    private Set<Subscriber> subscribers = new HashSet<>();
+
+    void register(Subscriber subscriber) {
+        subscribers.add(subscriber);
+    }
+
+    void onEvent(Event ev) {
+        for (Subscriber ob : subscribers) {
+            ob.notify(ev);
+        }
+    }
+}
+
+interface Subscriber {
+    void notify(Event ev);
```

Comment:
I believe that this should have some more logic, but not sure WHAT that is.

Reply:
Nope. Cannot (\* ok you can, but ignore that for now) add logic to an interface.